hera_sim/visibilities/vis_cpu.py

```diff
@@ -92,7 +124,13 @@ def validate(self):
                 * len(self.lsts)):
             raise ValueError("VisCPU requires that every baseline uses the "
                              "same LSTS.")
-
+        
+        # Check to make sure enough beams are specified
+        if not self.use_pixel_beams:
+            for ant in self.uvdata.get_ants():
+                assert len(np.where(self.beam_ids == ant)[0]), \
+                       "No beam found for antenna %d" % ant
+        
```

Comment:
Why do you need a separate beam instance for each antenna? Also, if that is really necessary, you should be able to test this more simply by checking `assert ant in self.beam_ids`

Reply:
PerturbedPolyBeams may be different for every antenna. Changed to use "ant in self.beam_ids".

Reply:
Great, thanks.